CHANGELOG.md

```diff
@@ -1,5 +1,94 @@
+
+```js
+const traceApi = require('@google-cloud/trace-agent').start();
+traceApi.runInRootSpan({ name: 'my-custom-span' }, (rootSpan) => {
+  if (!rootSpan) {
+    // This code path will no longer execute.
+  }
+  const childSpan = traceApi.createChildSpan({ name: 'my-custom-smaller-span' });
+  if (!childSpan) {
+    // This code path will no longer execute.
+  }
+  // ...
+});
+```
+
+Instead, "phantom" `SpanData` will be returned, which expose an identical, but non-functional API to that of a "real" `SpanData` object. The purpose of this change is to alleviate the burden of having branching code paths based on whether your code was being traced or not -- you may now assume that functions such as `addLabel` and `endSpan` are _always_ on the given `SpanData` object.
+
+If you _must_ execute a separate code path based on whether your code is being traced or not, you may now use `traceApi.isRealSpan`:
+
+```js
+const traceApi = require('@google-cloud/trace-agent').start();
+traceApi.runInRootSpan({ name: 'my-custom-span' }, (rootSpan) => {
+  if (!traceApi.isRealSpan(rootSpan)) {
+    // Some code that should be executed because this request is not being traced.
+  }
+  const childSpan = traceApi.createChildSpan({ name: 'my-custom-smaller-span' });
+  if (!traceApi.isRealSpan(childSpan)) {
+    // Alternatively, you may directly check whether a span is untraced or uncorrelated.
+    if (childSpan.type === traceApi.spanTypes.UNCORRELATED) {
+      // Some code that should be executed because we lost context.
+    } else if (childSpan.type === traceApi.spanTypes.UNTRACED) {
+      // Some code that should be executed because the request was not sampled
+      // (or otherwise disallowed by the tracing policy).
+    }
+  }
+  // ...
+});
+```
+
+This affects both plugins and the custom tracing API. All built-in tracing plugins have been changed correspondingly.
+
+#### Log Messages
+
+All logging output has been revised for consistency, and now include class/function names, as well as parameters in `[square brackets]`.
+
+#### Plugin Loader
+
+The plugin loading mechanism has been completely re-written. There should be no observable changes (please file an [issue](https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/issues) if you encounter one.) The re-write fixes an issue where plugins can't patch modules with circular dependencies correctly ([#618](https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/issues/618)).
+
+#### Tracing `pg`
+
+We now support tracing for `pg` versions 6 - 7.
+
+For consistency between tracing version 6 and 7, span labels in `pg` (when `config.enhancedDatabaseReporting` is enabled) now contain pre-processed query values rather than post-processed values.
```

Comment:
Side observation (should not affect this PR): would it make sense to have plugin specific documentation pages with information like this?

Reply:
Good idea, I've opened #709 to track this.